diplom.py

```diff
@@ -53,32 +61,51 @@ def mean_average_precision(predicted_probability_distribution,
 # print test_X[0], test_y[0]
 
 from sklearn import linear_model
-logistic = linear_model.LogisticRegression(C = 1, solver = 'lbfgs', 
-    multi_class = 'multinomial', max_iter = 100)
-print('LogisticRegression score: %f'
-      % logistic.fit(train_X, train_y).score(test_X, test_y))
+from sklearn.ensemble import RandomForestClassifier
+if args.model == "logistic_regression":
+    logistic = linear_model.LogisticRegression(C = 1, solver = 'lbfgs', 
+                                               multi_class = 'multinomial', max_iter = 100)
+else:
+    logistic = RandomForestClassifier(n_estimators=10)
+    
+logistic.fit(train_X, train_y)
+# print('LogisticRegression score: %f'
+#       % logistic.fit(train_X, train_y).score(test_X, test_y))
 
 predicted_probability_distribution_test = logistic.predict_proba(test_X)
 predicted_probability_distribution_train = logistic.predict_proba(train_X)
 
+classes_by_probability = []
+k = -1
+for row in predicted_probability_distribution_test:
+    k += 1
+    classes_by_prob = []
+    for i in xrange(0, len(logistic.classes_)):
+        classes_by_prob.append([row[i], logistic.classes_[i]])
+    classes_by_prob.sort(reverse=True)    
+    for i in xrange(0, len(logistic.classes_)):
+        classes_by_prob[i] = classes_by_prob[i][1]
+    classes_by_probability.append(classes_by_prob)
+
+print metrics.mapk([[y] for y in test_y], classes_by_probability, k=5)
 
 MAP_test = mean_average_precision(predicted_probability_distribution_test, 
-                                  logistic.classes_, 
-                                  test_y)
+                                   logistic.classes_, 
```

Comment:
abnormal indent, revert this change please (probably you have used some tabs again).